packages/apollo-link-error/src/index.ts

```diff
@@ -30,36 +31,67 @@ export const onError = (errorHandler: ErrorHandler): ApolloLink => {
   return new ApolloLink((operation, forward) => {
     return new Observable(observer => {
       let sub;
+      let retrySub;
+      let retrying = false;
+
       try {
         sub = forward(operation).subscribe({
           next: result => {
             if (result.errors) {
-              errorHandler({
+              const retryForward = errorHandler({
```

Comment:
`retryForward` seems like a confusing name. Something more aligned with what the return result is would be appropriate. Maybe `overridingObservable` or `retriedResult`?

Reply:
fixed